app/model/Payment/ReadModel/QueryHandlers/RepaymentCandidateListQueryHandler.php

```diff
@@ -47,7 +47,7 @@ public function __invoke(RepaymentCandidateListQuery $query): array
         $repayments = array_map([DTO\RepaymentCandidateFactory::class, 'create'], $payments);
 
         $group = $this->groups->find($query->getGroupId());
-        if ($group->getObject()->getType()->equals(Type::CAMP())) {
+        if ($group->getObject() !== null && $group->getObject()->getType()->equals(Type::CAMP())) {
```

Comment:
Přesuňme pls `$group->getObject()` do proměnné, ať tady těžíme z analýzy.

Reply:
Protože tohle počítá s tím, že je `getObject()` pure (což nemáme nijak vynucený), takže statická analýza vidí:
`if ((SkautisEntity|null) !== null && (SkautisEntity|null)->getType()...`

Když to přesuneš do proměnné, tak se ty typy zúží:
`if (SkautisEntity|null) !== null && (SkautisEntity)->getType()` což je to co chceme - umožní nám to pak tu analýzu zestriktnit.

Reply:
upraveno